js/src/dapps/dappreg/DappModal/dappModal.js

```diff
@@ -0,0 +1,412 @@
+import React, { Component, PropTypes } from 'react';
+
+import { api } from '../parity';
+import DappsStore from '../dappsStore';
+import Button from '../Button';
+import Input from '../Input';
+import Modal from '../Modal';
+import ModalDelete from '../ModalDelete';
+import ModalUpdate from '../ModalUpdate';
+import SelectAccount from '../SelectAccount';
+
+import styles from './dappModal.css';
+
+@observer
+export default class DappModal extends Component {
+  static propTypes = {
+    dapp: PropTypes.object.isRequired,
+    open: PropTypes.bool.isRequired,
+    onClose: PropTypes.func.isRequired
+  };
+
+  state = {
+    showDelete: false,
+    showUpdate: false,
+    updates: null,
+    updating: false
+  };
+
+  dappsStore = DappsStore.instance();
+
+  render () {
+    const { dapp, open } = this.props;
+    const { showDelete, showUpdate, updates } = this.state;
+
+    if (!open) {
+      return null;
+    }
+
+    return (
+      <div>
+        {
+          showDelete
+          ? (
+            <ModalDelete
+              dappId={ dapp.id }
+              onClose={ this.handleDeleteClose }
+              onDelete={ this.handleDeleteConfirm }
+            />
+          )
+          : null
+        }
+        {
+          showUpdate
+          ? (
+            <ModalUpdate
+              dappId={ dapp.id }
+              onClose={ this.handleUpdateClose }
+              onConfirm={ this.handleUpdateConfirm }
+              updates={ updates }
+            />
+          )
+          : null
+        }
+
+        <Modal
+          header={ this.renderHeader(dapp) }
+          onClose={ this.handleClose }
+        >
+          { this.renderContent(dapp) }
+        </Modal>
+      </div>
+    );
+  }
+
+  renderContent (dapp) {
+    const manifest = dapp.manifest.content || {};
+
+    return (
+      <div>
+        <div>
+          { this.renderInputs(dapp) }
+        </div>
+
+        { this.renderActions(dapp) }
+
+        <div className={ styles.code }>
+          <div className={ styles.codeTitle }>manifest.json</div>
+          <div className={ styles.codeContainer }>
+            <code>{ JSON.stringify(manifest, null, 2) }</code>
+          </div>
+        </div>
+      </div>
+    );
+  }
+
+  renderActions (dapp) {
+    if (!dapp.isOwner) {
+      return null;
+    }
+
+    const { isEditing } = dapp;
+    const { updating } = this.state;
+
+    if (updating) {
+      return (
+        <div className={ styles.actions }>
+          <Button
+            className={ styles.button }
+            disabled
+            label='Updating...'
+          />
+        </div>
+      );
+    }
+
+    if (isEditing) {
+      return (
+        <div className={ styles.actions }>
+          <Button
+            className={ styles.button }
+            disabled={ !dapp.canSave }
+            label='Save'
+            onClick={ this.handleSave }
+          />
+          <Button
+            className={ styles.button }
+            label='Fetch Registry'
```

Comment:
Ahhh, that one. Ok, will ponder a bit on a better more explicit name, it escapes me at present as well...